### PR TITLE
Parse ambiguous timezones

### LINKED ISSUE #447: Ambiguous timezone abbreviations are incorrectly interpreted for DST dates
There are several timezones that refer to both daylight savings time (DST) and non-DST timezones, depending on the date in question. For example, Eastern Time Zone (ET) refers to Eastern Standard Time (EST) during standard time, and Eastern Daylight Time (EDT) during daylight saving time. Chrono matches two such timezones: ET and PT. Other examples include Central Time (CT), Mountain Time (MT) and Central European Time (CET).

CET is kind of a special case, since "many sources define CET as a constant UTC+1. In common usage however, CET usually refers to the time observed in most of Europe, be it standard time or daylight saving time." ([source](https://time.is/CET)). In other words, CET refers to CET (UTC+1) during the non-DST part of the year, and it refers to CEST (UTC+2) during the DST part of the year.

Currently, Chrono always interprets ET as EST and PT as PST. During DST, this interpretation is usually wrong. Therefore, I suggest the following procedure:

1. Determine whether DST should be used based on the parsed date, if possible.
2. Otherwise, use the reference date to determine whether to use DST.
3. If DST should be used, return `tzOffset` corresponding to the DST times (EDT when "ET" is parsed, PDT when "PT" is parsed).
3. Otherwise, return `tzOffset` corresponding to the standard times (EST for "ET", PST for "PT").

The change does involve making some assumptions, but although not perfect I believe it will be better than the current implementation. I can submit a PR implementing this, but I first wanted to post here to get input and see if everyone agrees with my suggestion.

### PR DESCRIPTION
**Pull request by JoakimNil**
Closes #447. I changed my mind, and decided that a PR is probably the best way to focus the discussion 😄

Please see the note in #447 regarding CET. There were some existing tests that expected CET to be 60 minutes, even during DST. I changed those tests to reflect the nuance in interpreting CET. I expect there will be cases where people expect CET to be 60 minutes even during DST, but I firmly believe that it's more common to expect CET to be 120 minutes during DST.

I also made some structural changes: A lot of the timezone logic was duplicated in `timezone.ts` and `ExtractTimezoneAbbrRefiner.ts`, so I moved most of it to `timezone.ts`.

Finally, there's a drive-by commit to export ParsingResult, which is commonly needed by chrono users.

### CODE REVIEW

**Comment by wanasit** [comment]
Hello @JoakimNil! Thanks for your great work! I think your timezone change is pretty nice and sophisticated. 

Could you please reply to my code review comments before I merge it?

**Comment by JoakimNil** [comment]
@wanasit thanks for taking the time to follow up this PR. I can't seem to find your comments. Perhaps they are still pending in an unsubmitted review?

**Comment by wanasit** [comment]
@JoakimNil Ops. Sorry I am not really familiar with GitHub review 😓

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { DebugHandler, DebugConsume } from "./debugging";
 import * as en from "./locales/en";
 import { Chrono, Parser, Refiner } from "./chrono";
+import { TimezoneAbbrMap } from "./timezone";
+import type { ParsingResult } from "./results";
 
-export { en, Chrono, Parser, Refiner };
+export { en, Chrono, Parser, Refiner, TimezoneAbbrMap, ParsingResult };
```
[the remark this answers is not in the document]
Comment (JoakimNil) [reply src/index.ts]: Please see the newest changes. The changes are mostly according to your suggestion, and there's now better documentation :)

However, I kept the name `AmbiguousTimezoneMap` because I feel like it better describes the uncertainty associated with parsing these kinds of timezones. Moreover, I feel like "WithDstInfo" may incorrectly give the impression that these timezones are like normal timezone mappings, but with additional information. On the contrary, the use of these mappings reduces the certainty of the parsing due to the DST ambiguity.

File: src/index.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { DebugHandler, DebugConsume } from "./debugging";
 import * as en from "./locales/en";
 import { Chrono, Parser, Refiner } from "./chrono";
+import { TimezoneAbbrMap } from "./timezone";
+import type { ParsingResult } from "./results";
```
[the remark this answers is not in the document]
Comment (JoakimNil) [reply src/index.ts]: yes, it wasn't appropriate for this PR and the change has been removed.

**Comment by JoakimNil** [comment]
@wanasit I *finally* got around to addressing your suggestions. I think the PR is significantly improved after taking them into account.

**Comment by wanasit** [comment]
@JoakimNil Thanks for working on this. I added a few additional comments, but it's a significant improvement as you said. 

If you don't have time to address the comments, I'll still merge it later this week.

**Comment by JoakimNil** [comment]
@wanasit sure, I have some spare time. I'm not seeing the new comments though?

**Comment by wanasit** [comment]
Ops. I forgot to click send the review again.

File: test/de/de_time_exp.test.ts [hunk 3]
```diff
@@ -180,22 +180,22 @@ test("Test - Range Expression", function () {
...
         expect(result.text).toBe(text);
         expect(result.start.isCertain("timezoneOffset")).toBe(true);
         expect(result.start.get("timezoneOffset")).toBe(60);
     });
 
-    testSingleCase(chrono.de, "14 Uhr cet", new Date(2016, 3, 28), (result, text) => {
+    testSingleCase(chrono.de, "14 Uhr cet", new Date(2016, 4, 28), (result, text) => {
         expect(result.text).toBe(text);
         expect(result.start.isCertain("timezoneOffset")).toBe(true);
-        expect(result.start.get("timezoneOffset")).toBe(60);
```
[the remark this answers is not in the document]
Comment (JoakimNil) [reply test/de/de_time_exp.test.ts:198]: Without changing the test, some of them will fail after the change. This is because they expect a date during DST with CET timezone to have an offset of 60 minutes, while in fact it should be 120 minutes. I refer to my previous notes (here and in #447) for more details regarding CET offsets.

I also changed a few tests that weren't incorrectly formulated and thus didn't strictly need to be changed. However, they all had really unfortunate reference dates that I think made them hard to understand.

The reference date March 28th happens during DST in some years, and outside of DST in other years. So to understand why these tests should result in offsets of 60 and not 120 requires the developer to be extremely conscious of the nuances of how ambiguous timezones are parsed. Testing ambiguous timezone edge cases is not the purpose of these tests. Therefore, I think it's better to keep those tests in en_timezone_exp.test.ts, and instead change these as I have done to make them easier to interpret for their intended purpose.

**Comment by JoakimNil** [comment]
@wanasit I've now addressed your comments with a few new commits. Also see my reply regarding tests, which I didn't address with any code changes. Would you mind taking a look at the new changes?

**Comment by wanasit** [comment]
Cool! thanks again for working on this!.
